src/components/Pagination/pagination-utils.ts

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+
+export function getLinkDataCy(index: number, totalPages: number): string {
+  switch (index) {
+    case 0:
+      return 'prevLink';
+    case totalPages + 1:
+      return 'nextLink';
+    default:
+      return 'pageLink';
+  }
+}
+
+export function getLinkHref(index: number, totalPages: number): string {
+  switch (index) {
+    case 0:
+      return '#prev';
+    case totalPages + 1:
+      return '#next';
+    default:
+      return `#${index}`;
+  }
+}
+
+export function getLinkTextContent(index: number, totalPages: number): string {
+  switch (index) {
+    case 0:
+      return '«';
+    case totalPages + 1:
+      return '»';
+    default:
+      return `${index}`;
+  }
+}
+
+export function handleClickPagination(
+  event: React.MouseEvent<HTMLUListElement>,
+  currentPage: number,
+  pagesAmount: number,
+  onPageChange: (page: number) => void,
+): void {
+  const target = event.target as HTMLElement;
+  const dataCYValue = target.getAttribute('data-cy');
+  const currentPageNumber: number = Number(target.textContent);
```

Comment:
Converting `textContent` to a number assumes that the `textContent` is always a valid number. You should add a check to ensure that the conversion is successful and handle cases where it is not.